pkg/io/jsonio/unmarshalConns.go

```diff
@@ -0,0 +1,158 @@
+	if isSG {
+		res, err = defs.LookupForSGSynth(resourceType, resource.Name)
+		updateBlockedResourcesSGSynth(blockedResources, res)
+	} else {
+		res, err = defs.LookupForACLSynth(resourceType, resource.Name)
+		updateBlockedResourcesACLSynth(blockedResources, res)
+	}
+	return res, resourceType == ir.ResourceTypeExternal, err
+}
+
+func translateProtocols(protocols spec.ProtocolList) ([]*ir.TrackedProtocol, error) {
+	var result = make([]*ir.TrackedProtocol, len(protocols))
+	for i, _p := range protocols {
+		res := &ir.TrackedProtocol{Origin: protocolOrigin{protocolIndex: i}}
+		switch p := _p.(type) {
+		case spec.AnyProtocol:
+			if len(protocols) != 1 {
+				log.Println("when allowing any protocol, there is no need in other protocols")
+			}
+			res.Protocol = netp.AnyProtocol{}
+		case spec.Icmp:
+			icmp, err := netp.ICMPFromTypeAndCode(p.Type, p.Code)
+			if err != nil {
+				return nil, err
+			}
+			res.Protocol = icmp
+		case spec.TcpUdp:
+			tcpudp, err := netp.NewTCPUDP(p.Protocol == spec.TcpUdpProtocolTCP, p.MinSourcePort, p.MaxSourcePort,
+				p.MinDestinationPort, p.MaxDestinationPort)
+			if err != nil {
+				return nil, err
+			}
+			res.Protocol = tcpudp
+		default:
+			return nil, fmt.Errorf("impossible protocol: %v", p)
+		}
+		result[i] = res
+	}
+	return result, nil
+}
+
+func translateResourceType(defs *ir.Definitions, resource *spec.Resource) (ir.ResourceType, error) {
+	switch resource.Type {
+	case spec.ResourceTypeExternal:
+		return ir.ResourceTypeExternal, nil
+	case spec.ResourceTypeSubnet:
+		return ir.ResourceTypeSubnet, nil
+	case spec.ResourceTypeNif:
+		return ir.ResourceTypeNIF, nil
+	case spec.ResourceTypeInstance:
+		return ir.ResourceTypeInstance, nil
+	case spec.ResourceTypeVpe:
+		return ir.ResourceTypeVPE, nil
+	case spec.ResourceTypeSegment:
+		if _, ok := defs.SubnetSegments[resource.Name]; ok {
+			return ir.ResourceTypeSubnetSegment, nil
+		}
+		if _, ok := defs.CidrSegments[resource.Name]; ok {
+			return ir.ResourceTypeCidrSegment, nil
+		}
+		if _, ok := defs.NifSegments[resource.Name]; ok {
+			return ir.ResourceTypeNifSegment, nil
+		}
+		if _, ok := defs.InstanceSegments[resource.Name]; ok {
+			return ir.ResourceTypeInstanceSegment, nil
+		}
+		if _, ok := defs.VpeSegments[resource.Name]; ok {
+			return ir.ResourceTypeVpeSegment, nil
+		}
+		return ir.ResourceTypeSubnet, fmt.Errorf("could not find segment %v", resource.Name)
+	}
+	return ir.ResourceTypeSubnet, fmt.Errorf("unsupported resource type %v (%v)", resource.Type, resource.Name)
+}
+
+func updateBlockedResourcesSGSynth(blockedResources *ir.BlockedResources, resource *ir.ConnectedResource) {
+	for _, namedAddrs := range resource.CidrsWhenLocal {
+		if _, ok := blockedResources.BlockedInstances[namedAddrs.Name]; ok {
+			blockedResources.BlockedInstances[namedAddrs.Name] = false
+		}
+		if _, ok := blockedResources.BlockedVPEs[namedAddrs.Name]; ok {
+			blockedResources.BlockedVPEs[namedAddrs.Name] = false
+		}
```

Comment:
shouldn't we check `resource.ResourceType` (for the rare case where we have an Instance and a VPE with the same name?